deploy/charts/rook-ceph/values.yaml

```diff
@@ -514,6 +514,18 @@ csi:
     # - topology.kubernetes.io/zone
     # - topology.rook.io/rack
 
+  readAffinity:
+    # -- Enable read affinity for rbd volumes.
+    enabled: false
+    # -- crushLocationLabels define which node labels to use
+    # as CRUSH location. This corresponds to the values used in
+    # ceph CRUSH map.
+    # @default -- labels listed [here](../CRDs/Cluster/ceph-cluster-crd.md#osd-topology)
+    crushLocationLabels:
+    # - kubernetes.io/hostname
```

Comment:
> This is probably a bad example since https://rook.io/docs/rook/latest/CRDs/Cluster/ceph-cluster-crd/#osd-topology says:
> 
> > Note that the `host` is added automatically to the hierarchy by Rook. The host cannot be specified with a topology label.
> 
> Also, `hostname` is not a valid CRUSH type (at least not by default).

Thanks for pointing it out, I've modified code to handle it in cephcsi pr.

Reply:
Where would `kubernetes.io/hostname` label come from?  I don't see it in `CRUSHTopologyLabels` array.

Reply:
> Where would `kubernetes.io/hostname` label come from? I don't see it in `CRUSHTopologyLabels` array.

I've added to the default list now.

Reply:
My question was deeper than that: who is expected to set it?  IIUC these labels are supposed to be set automatically by Rook but it doesn't look like that would be the case for this `kubernetes.io/hostname` label.

Reply:
> My question was deeper than that: who is expected to set it? IIUC these labels are supposed to be set automatically by Rook but it doesn't look like that would be the case for this `kubernetes.io/hostname` label.

kubernetes.io/hostname should be set by kubernetes itself. 

The other labels should to set the user/operator(for ex ocs-operator) which creates storage cluster. 
https://rook.io/docs/rook/v1.10/CRDs/Cluster/ceph-cluster-crd/#osd-topology 

`
The topology of the cluster is important in production environments where you want your data spread across failure domains. The topology can be controlled by adding labels to the nodes. When the labels are found on a node at first OSD deployment, Rook will add them to the desired level in the CRUSH map.
`

Reply:
Is the value of `kubernetes.io/hostname` label (that Ceph CSI would use when specifying `crush_location` on the client side) guaranteed to be the same as the `host` value that gets set in the CRUSH map on the server side by Rook?

Reply:
Yes, the hostname label will be used for the CRUSH host as seen [here](https://github.com/rook/rook/blob/b5ebb2ed972fb03d57cd8b18174ebf3159b4b296/pkg/operator/ceph/cluster/osd/topology.go#L102-L105).